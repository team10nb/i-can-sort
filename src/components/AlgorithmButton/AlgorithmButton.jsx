/*
    Author: Yuting Jiang

    Algorithms code is on the button. Click to run the algorithm.
*/

import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from "@material-ui/core/Typography";
import backgroundImage from "../../resource/run.png";

export default function AlgorithmButton(props) {
    const { title, code, paddingLeft, paddingRight, onClick } = props;
    const styles = {
        div: {
            textAlign: "left",
            paddingLeft: paddingLeft,
            paddingRight: paddingRight,
        },
    };

    const useStyles = makeStyles(() => ({
        buttonBase: {
            paddingLeft: 20,
            width: 250,
            height: 270,
            borderRadius: 15,
            justifyContent: "left",
            textAlign: "left",
            "&:hover": {
                backgroundImage: `url(${backgroundImage})`,
            },
        },
    }));
    const classes = useStyles();

    return (
        <div style={styles.div}>
            <ButtonBase className={classes.buttonBase} onClick={onClick}>
                <Typography variant="body2" align="left">
                    <span>{title}</span>
                    <span>{code}</span>
                </Typography>
            </ButtonBase>
        </div>
    );
}
