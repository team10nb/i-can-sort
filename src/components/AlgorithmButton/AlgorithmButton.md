### **Developer**

Yuting Jiang

Reference: MATERIAL-UI (https://material-ui.com)

###  **Introduction**

Used in Correctness-Tutoial-Example page.
To Show an example algorithm.

###  **Import**

```html
import { makeStyles } from "@material-ui/core/styles";
import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from "@material-ui/core/Typography";
```

###  **Props**

| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
| title | string |         | Title of the algorithm button            |
| code | string |         | Code of the algorithm           |
| paddingLeft | number |    0     | The distance to its left element           |
| paddingRight | number |    0     | The distance to its right element                |
| onClick | fn |         | Handle onClick event            |

###  **Demo**

```jsx
const title_interminate =
    <h3>Example1<br />
    </h3>

const code_interminate =
    <pre style={{ marginTop: -20 }}>
        {`
factorial (int n){
  int result = 1;
  for (int i=1; ; i++){
    result = result * i;
  }
  return result;
}
`}
    </pre>;

<AlgorithmButton title={title_interminate} code={code_interminate} paddingLeft={0} paddingRight={0} />
```