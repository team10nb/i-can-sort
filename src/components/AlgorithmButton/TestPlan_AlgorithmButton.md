#### Component: AlgorithmButton

|Test Name|Description|Operation|Expected Outcome|
|---------|-----------|---------|----------------|
|should contain title and code|||title and code should be shown on the button|
|should call onClick||Click AlogrithmButton|onClick will be called once|
|should match Snapshot|Compare new snapshot with old one|  | Snapshots are the same|