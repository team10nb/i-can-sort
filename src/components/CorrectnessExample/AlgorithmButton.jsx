import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ButtonBase from '@material-ui/core/ButtonBase';
import Typography from '@material-ui/core/Typography';
import backgroundImage from '../../Resource/run.png';
// const color = '#FEE5D4'

export default function AlgorithmButton(props) {
  const {title, code, paddingLeft, paddingRight, onClick, } = props;
  const styles = {
    div:{
      textAlign: 'left',
      paddingLeft:paddingLeft,
      paddingRight:paddingRight,
    },
  };

  const useStyles = makeStyles((theme) => ({
    buttonBase: {
        width: 260,
        height: 260,
        borderRadius: 15,
        justifyContent: 'left',
        textAlign: 'left',
        '&:hover':{
            backgroundImage: `url(${backgroundImage})`,
        },
    }
  }));
  const classes = useStyles();

  return (
      <div style={styles.div}> 
          <ButtonBase className={classes.buttonBase} onClick={onClick}>
            <Typography variant="body2" align='left'>
                {title}
                {code}
            </Typography> 
          </ButtonBase>
      </div>
  );
}
