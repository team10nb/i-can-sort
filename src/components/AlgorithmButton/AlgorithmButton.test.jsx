/*
    Author: Yuting Jiang
*/

import { render } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import AlgorithmButton from './AlgorithmButton';

test('should contain title and code', () => {
    
    const mockHandleClick = jest.fn();
    const title = "title";
    const code = "code";

    const {getByText} = render(<AlgorithmButton title={title} code={code} onClick={mockHandleClick} />);
    getByText("title");
    getByText("code");

})

test('should call onClick', () => {
    
    const mockHandleClick = jest.fn();
    const title = "title";

    const {getByText} = render(<AlgorithmButton title={title} onClick={mockHandleClick} />);
    const button = getByText("title");
    userEvent.click(button);
    expect(mockHandleClick).toBeCalledTimes(1);

})


test('should match Snapshot', () => {
    
    const mockHandleClick = jest.fn();
    const title = "title";
    const code = "code";

    const content = render(<AlgorithmButton title={title} code={code} onClick={mockHandleClick} />);
    expect(content).toMatchSnapshot();

})

