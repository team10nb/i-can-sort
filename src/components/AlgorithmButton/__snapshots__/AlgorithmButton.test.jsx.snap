// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="text-align: left;"
      >
        <button
          class="MuiButtonBase-root makeStyles-buttonBase-3"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft"
          >
            <span>
              title
            </span>
            <span>
              code
            </span>
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="text-align: left;"
    >
      <button
        class="MuiButtonBase-root makeStyles-buttonBase-3"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft"
        >
          <span>
            title
          </span>
          <span>
            code
          </span>
        </p>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
